installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -250,6 +307,16 @@ public static ActionResult StateDeploy(Session session)
         {
             ActiveState.RollbarHelper.ConfigureRollbarSingleton();
 
+            if (!Environment.Is64BitOperatingSystem)
+            {
+                Record record = new Record();
+                record.FormatString = "This installer cannot be run on a 32-bit operating system";
+
+                ActiveState.RollbarHelper.Report(record.FormatString);
+                session.Message(InstallMessage.Error | (InstallMessage)MessageBoxButtons.OK, record);
+                return ActionResult.Failure;
+            }
```

Comment:
Why wait till the deploy step to do this?

Reply:
The deploy CA is where the install logic starts, it calls `InstallStateTool` and then carries on with the deployment. The only custom actions that run before this are to reset the progress bar and to setup the rollback information which could arguably also have this check